client-linuxapp/src/serviceinfo.rs

```diff
@@ -358,7 +368,8 @@ async fn process_serviceinfo_in(si_in: &ServiceInfo, si_out: &mut ServiceInfo) -
     let mut rhsm_activation_key: Option<String> = None;
     let mut rhsm_perform_insights: Option<bool> = None;
 
-    let mut binary_file_in_progress = BinaryFileInProgress::new();
+    let binary_file_prefix = env::var("BINARYFILE_PATH_PREFIX")?;
+    let mut binary_file_in_progress = BinaryFileInProgress::new(Some(binary_file_prefix.clone()));
```

Comment:
Would it be possible to pass it in as a `&str` instead, so we don't have to clone all the time?
Also, this seems to imply that `binary_file_prefix` is never empty?
Maybe `let binary_file_prefix = env::var("...").ok()`, to make it an `Option`, and then pass it in as `Option<&str>`, or even `Option<Path>`?

Reply:
I'll check for emptiness - as for having a &str instead of a String it's messy afaict but I'll see what I can come up with (lifetime and Some(&str) being messy too etc etc)

Reply:
fixed